### DISCUSSION TITLE
Teltonika RMS not working with AGH

### DISCUSSION THREAD

**Discussion by FilipQSense** [AdguardTeam/AdGuardHome#7224] [Q&A]
After installing AdGuard Home on Teltonika routers (RUTX12 and RUTX50) connecting to Telto RMS (Remote Management System) results in failure: "Could not resolve hostname". I tried disabling all filters and tried excluding RMS website from being blocked through non-standard rules to no success. After uninstalling AGH the issue goes away immediately. 
RMS uses ports beetwen 15009-15100 for reaching the platform and ports over 20000 for remote connection to the device so I don't think the issue lies in ports being used by AGH.
I can access the website using a device connected to network with AdGuard working, it's just the router itself that can't reach it.
Is there anything more I can try?
[rms.teltonika-networks.com](https://rms.teltonika-networks.com/)

**Comment by reggyralfs** [comment]
Did you ever find a solution to this issue?

**Reply by FilipQSense** [reply]
I've only managed to narrow it down, the moment I set AGH as DNS server RMS stops working. I just assumed that I had to give up one or the other so it basically killed my plan of implementing AGH on all of my routers.

**Reply by reggyralfs** [reply]
Oh that's not good, I just purchased a RUTX with the intention of running AGH as the main use.   Fingers crossed I have more luck!

**Reply by FilipQSense** [reply]
I hope you do :)
I'm sure there is something I haven't tried yet so if you happen to find the fix for this, please let me know :)

**Comment by reggyralfs** [comment]
I did get the 2 working harmoniously. 
Left RUTX alone with DNS on 53
Setup AGH to run on port 5353
Used IPTABLES to route all requests to 53 to 5353
iptables -t nat -A PREROUTING -i br-lan -p udp --dport 53 -j DNAT --to 10.0.0.1:5353
iptables -t nat -A PREROUTING -i br-lan -p tcp --dport 53 -j DNAT --to 10.0.0.1:5353

**Reply by ChriWoPS** [reply]
@reggyralfs
Can you please tell me which instructions you have used to install Adguard on the RUTX50?
Thank you in advance

**Comment by reggyralfs** [comment]
Apologies I don't have a direct set of instructions as I cobbled together the installation from multiple sources, and only kept limited notes!

Here is what I have:

**Install:**
opkg -e /etc/opkg/openwrt/distfeeds.conf update 
opkg -e /etc/opkg/openwrt/distfeeds.conf install adguardhome

**Once AG is setup and running taking DNS requests on port 5353:**
iptables -t nat -A PREROUTING -i br-lan -p udp --dport 53 -j DNAT --to 10.0.0.1:5353
iptables -t nat -A PREROUTING -i br-lan -p tcp --dport 53 -j DNAT --to 10.0.0.1:5353
In my case I was using the 10.0.0.0/24 subnet, adjust to yours.

**To have AdGuard Home automatically start on boot and start the service::**
service adguardhome enable
service adguardhome start

**To check logs:**
Logread -f

**Note:**
Running too many blocklists will consume too much RAM, keep at least 50-60mb RAM free otherwise AG can consume all RAM and crash the service.  I did also setup a nightly reboot at 3am to flush memory, and set AG logging to 24hrs only.

**Reply by ChriWoPS** [reply]
Thank you this helps
